docs/pages/2020_BIGMAP/sections/qe.rst

```diff
@@ -323,7 +323,141 @@ Use this to produce a PDF of the band structure:
+    Info: to show groups of all types, use the `-a/--all` option.
+      PK  Label            Type string    User
+    ----  ---------------  -------------  ---------------
+       6  tutorial_pbesol  core           aiida@localhost
+       7  tutorial_lda     core           aiida@localhost
+       8  tutorial_pbe     core           aiida@localhost
+
+Image you want to understand the influence of the functional on the magnetization of the structure.
+Let's *build* a query that helps us investigate this question.
+Start the ``verdi shell``, and load the ``StructureData`` and ``PwCalculation`` classes:
+
+.. code-block:: ipython
+
+    In [1]: StructureData = DataFactory('structure')
+       ...: PwCalculation = CalculationFactory('quantumespresso.pw')
+
+We start every query by creating an instance of the ``QueryBuilder`` class:
+
+.. code-block:: ipython
+
+    In [2]: qb = QueryBuilder()
+
+To build a query, we *append* entities (nodes, groups, ...) to the query.
+Let's build the query for one of the groups - say, ``tutorial_pbesol`` - step by step to help understand the process.
+We first append the ``Group`` to our ``QueryBuilder`` instance:
+
+.. code-block:: ipython
+
+    In [3]: qb.append(Group, filters={'label': 'tutorial_pbesol'}, tag='group');
+
+Let's explain the different arguments used in this call of the ``append()`` method:
+
+    * The first *positional* argument is the ``Group`` class, preloaded in the ``verdi shell``.
+    * The first *keyword* argument is ``filters``, here we *filter* for the group with ``label`` equal to ``tutorial_pbesol``.
+    * The second *keyword* argument is ``tag``.
+      This is reference we use to indicate *relationships* between nodes in future ``append()`` calls.
+
+Next, we'll look for all the ``PwCalculations`` in this group:
+
+.. code-block:: ipython
+
+    In [4]: qb.append(PwCalculation, with_group='group', tag='pw');
+
+Here, we use the ``'group'`` tag we created in the previous step to query for ``PwCalculation``'s in the ``tutorial_pbesol`` group using the ``with_group`` *relationship* argument.
+Moreover, we once again *tag* this append step of our query with ``pw``.
+Let's have a look at how many nodes we have in the ``tutorial_pbesol`` group:
+
+.. code-block:: ipython
+
+    In [5]: qb.count()
+    Out[5]: 57
+
+Great, now let's figure out which structures are magnetic!
+Of course, the information we are interested in are the structures and their absolute magnetization, which we'll query for in the final two steps.
+First, we'll *append* the ``StructureData`` to the query:
+
+.. code-block:: ipython
+
+    In [6]: qb.append(StructureData, with_outgoing='pw', project='extras.formula');
+
+In this step, we've used the ``with_outgoing`` relationship to look for structures that have an *outgoing* link to the ``PwCalculations`` referenced with the ``pw`` tag.
+That means that from the ``PwCalculation``'s perspective, the ``StructureData`` is an input, exactly what we want.
```

Comment:
I don't think we are explicit before on how exactly is that we are going to do the queries, no? I.e. we never specify that in a middle step we will want to have the structure of the PwCalcjobs.

```suggestion
That means that from the ``PwCalculation``'s perspective, the ``StructureData`` is an input.
```